lib/google_scraping/dashboard/keywords.ex

```diff
@@ -42,15 +43,73 @@ defmodule GoogleScraping.Dashboard.Keywords do
   Creates keywords list.
   """
   def create_keyword_list(keyword_list, user) do
-    Enum.each(keyword_list, fn keyword_name ->
-      keyword_params = %{
-        user_id: user.id,
-        name: keyword_name,
-        status: :new
-      }
-
-      # TODO: start background jobs for each created keyword
-      create_keyword(keyword_params)
+    keywords_data =
+      Enum.map(keyword_list, fn keyword_name ->
+        %{
+          user_id: user.id,
+          name: keyword_name,
+          status: :new
+        }
+      end)
+
+    Repo.transaction(fn ->
+      try do
+        keyword_ids = validate_and_create_keywords_list(keywords_data)
+
+        Enum.each(keyword_ids, fn keyword_id ->
+          create_keyword_background_job(keyword_id)
+        end)
+      catch
+        :error -> Repo.rollback(:invalid_keywords)
+      end
+    end)
+  end
+
+  @doc """
+  Gets a single keyword.
+  """
+  def get_keyword_by_id!(id), do: Repo.get!(Keyword, id)
+
+  @doc """
+  Mark keyword as in_progress
+  """
+  def mark_as_in_progress(keyword) do
+    keyword
+    |> Keyword.in_progress_changeset()
+    |> Repo.update!()
+  end
+
+  @doc """
+  Mark keyword as completed
+  """
+  def mark_as_completed(keyword, attrs) do
+    keyword
+    |> Keyword.completed_changeset(attrs)
+    |> Repo.update!()
+  end
+
+  @doc """
+  Mark keyword as failed
+  """
+  def mark_as_failed(keyword) do
+    keyword
+    |> Keyword.failed_changeset()
+    |> Repo.update!()
+  end
+
+  defp create_keyword_background_job(keyword_id) do
+    %{"keyword_id" => keyword_id}
+    |> KeywordScraperWorker.new()
+    |> Oban.insert()
+  end
+
+  defp validate_and_create_keywords_list(keywords_data) do
+    Enum.map(keywords_data, fn keyword_attrs ->
+      case create_keyword(keyword_attrs) do
+        {:ok, %Keyword{id: keyword_id}} -> keyword_id
+        # if one of the keywords is invalid, raise an error
+        {:error, _} -> throw(:error)
```

Comment:
Should we throw the changeset this returns? Then we can use `Logger.error` to log the issue and help with debugging in the future 💭 
What do you think? 🤔

Reply:
I think it is good idea, it was handled here: https://github.com/Nihisil/nimble-certification-elixir/pull/41/commits/c59e3802e96bb09b2c667ece1d63d5a1eb278b45